test/boost/memtable_test.cc

```diff
@@ -88,6 +111,22 @@ SEASTAR_TEST_CASE(test_memtable_conforms_to_mutation_source) {
     });
 }
 
+SEASTAR_TEST_CASE(test_reverse_memtable_conforms_to_mutation_source) {
+    return seastar::async([] {
+        run_mutation_source_tests([](schema_ptr s, const std::vector<mutation>& partitions) {
+            auto mt = make_lw_shared<memtable>(s->make_reversed());
+
+            for (auto&& m : partitions) {
+                mt->apply(reverse(m));
+            }
+
+            logalloc::shard_tracker().full_compaction();
```

Comment:
Why is this needed?

Reply:
I followed by analogy your addition to the `test_memtable_conforms_to_mutation_source`:
```
commit 8978d0ba1aa5491aeef3a1c3be1edcc763f60702
Author: Tomasz Grabiec <tgrabiec@cloudius-systems.com>
Date:   Tue Aug 4 10:15:53 2015 +0200

    tests: Test memtable data survives full compaction

diff --git a/tests/memtable_test.cc b/tests/memtable_test.cc
index fc84aee20..543ac0ef8 100644
--- a/tests/memtable_test.cc
+++ b/tests/memtable_test.cc
@@ -20,6 +20,8 @@ SEASTAR_TEST_CASE(test_memtable_conforms_to_mutation_source) {
                 mt->apply(m);
             }
 
+            logalloc::shard_tracker().full_compaction();
+
             return mt->as_data_source();
         });
     });
```
Perhaps this is not needed.